src/validation.cpp

```diff
@@ -2533,7 +2533,7 @@ void CChainState::UpdateTip(const CBlockIndex* pindexNew)
         const CBlockIndex* pindex = pindexNew;
         for (int bit = 0; bit < VERSIONBITS_NUM_BITS; bit++) {
             WarningBitsConditionChecker checker(bit);
-            ThresholdState state = checker.GetStateFor(pindex, m_params.GetConsensus(), warningcache[bit]);
+            ThresholdState state = checker.GetStateFor(pindex, m_params.GetConsensus(), warningcache.at(bit));
```

Comment:
Unrelated.

Sometimes I wonder how safe it is to throw in random places in the consensus code and then continue the program. Too bad there is no asserting `at()`.

![Screenshot from 2022-04-25 17-03-14](https://user-images.githubusercontent.com/6399679/165116935-32ea6030-0142-4d87-b27f-3348e5c98ace.png)

Reply:
There is a compile-time erroring `at()` -- `std::get<N>(my_array)` won't compile if `N` is outside the range. I've got a patchset that switches versionbits to working that way, fwiw; needs a lot of tidying up yet though, and builds on these patches.

Reply:
Nice. TIL https://en.cppreference.com/w/cpp/container/array/get